modules/core/src/main/java/org/apache/ignite/internal/visor/tx/VisorTxTask.java

```diff
@@ -74,6 +73,9 @@
     /** */
     private static final long serialVersionUID = 0L;
 
+    /** */
+    private static final int DEFAULT_REDUCE_LIMIT = 5;
```

Comment:
The logic looks a bit flawed to me. You use different values for map and reduce limits if the limit isn't set explicitly. However, if it is, you use the same value. I think default values should be equal. Otherwise, we will do some computations that will be discarded at the end(either in _map_ or in _reduce_ methods)